sdk/azidentity/chained_token_credential_test.go

```diff
@@ -97,6 +97,37 @@ func TestChainedTokenCredential_GetTokenFail(t *testing.T) {
 	}
 }
 
+func TestChainedTokenCredential_GetTokenWithUnavailableCredentialInChain(t *testing.T) {
+	srv, close := mock.NewServer()
+	defer close()
+	srv.AppendResponse(mock.WithError(&CredentialUnavailableError{CredentialType: "MockCredential", Message: "Mocking a credential unavailable error"}))
```

Comment:
There is `srv.AppendError()`, did that not work?

Reply:
Oops! I must have skipped over that when I was using my intellisense. Removing it!